AppCenterData/AppCenterData/Internal/Client/MSDataOperationProxy.m

```diff
@@ -69,14 +69,20 @@ - (void)performOperation:(NSString *_Nullable)operation
     if ([self shouldAttemptRemoteOperationForDocument:cachedDocument]) {
       MSLogInfo([MSData logTag], @"Performing remote operation");
       remoteDocumentBlock(^(MSDocumentWrapper *_Nonnull remoteDocument) {
+
         // If a valid remote document was retrieved, update local store
         if (remoteDocument.error == nil) {
+          
+          // For delete operation, delete the document from local cache.
+          // Otherewise, for other online scenarios, the intended pending operation in the local store is nil.
+          NSString *localStorageOperation = [kMSPendingOperationDelete isEqualToString:(NSString *)operation] ? kMSPendingOperationDelete : nil;
+          
+          // Perform local storage operation.
           [self updateLocalStore:token
```

Comment:
This will not delete the operation from the cache. It will only create or update and entry and set the pending operation to DELETE or nil for other operations.